crates/trie/sparse/src/state.rs

```diff
@@ -760,6 +783,56 @@ impl<F: BlindedProviderFactory> SparseStateTrie<F> {
     }
 }
 
+/// Result of [`decode_proof_nodes`].
+#[derive(Debug)]
+struct DecodedProofNodes {
+    /// Filtered, decoded and sorted proof nodes.
+    nodes: Vec<(Nibbles, TrieNode)>,
+    /// Number of nodes in the proof.
+    total_nodes: u64,
+    /// Number of nodes that were skipped because they were already revealed.
+    skipped_nodes: u64,
+    /// Number of new nodes that will be revealed. This includes all children of branch nodes, even
+    /// if they are not in the proof.
+    new_nodes: usize,
```

Comment:
why is new_nodes usize and total_nodes u64? according to the logic, always new_nodes >= total_nodes. i guess it is bc new_nodes is used to reserve, why not everything usize?

Reply:
was just more convenient to return u64 because it's reported in metrics as u64, but think it's nicer to have consistency in types, so fixed